src/connection.ts

```diff
@@ -1052,7 +1052,7 @@ class Connection extends EventEmitter {
       throw new TypeError('The "config" argument is required and must be of type Object.');
     }
 
-    if (typeof config.server !== 'string') {
+    if (typeof config.server !== 'string' && !config.options!.connector) {
       throw new TypeError('The "config.server" property is required and must be of type string.');
```

Comment:
We should change the error message since this is not just verify server property and it is no long required.
perhaps we can change it to something looks like : "The "config.server" property is required and must be of type string if "config.options.connector" is not used "

@arthurschreiber Do you think this message looks ok?

Reply:
I'm in favor of leaving the error message as is. My line of thought is that bringing up the `config.options.connector` property to users not using it might end up being more confusing.

Reply:
Agreed! Chatted with Arthur, and he thinks that we should keep the old message as well.

Reply:
awesome! 👍